File: cpp/api/src/base/ServerAndApp.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include "ServerAndApp.h"

namespace cameo {

ServerAndApp::ServerAndApp() {
}

ServerAndApp::ServerAndApp(std::unique_ptr<Server>& server, std::unique_ptr<App>& app) :
	m_server(std::move(server)), m_app(std::move(app)) {
}

Server& ServerAndApp::getServer() {
	return *m_server.get();
}

bool ServerAndApp::hasApp() const {
	return m_app.get() != nullptr;
}

App& ServerAndApp::getApp() {
	return *m_app.get();
}

void ServerAndApp::terminate() {

	m_server->terminate();
	m_app->terminate();
}

}

File: java/api/src/main/java/eu/ill/cameo/api/base/ServerAndApp.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.api.base;

/**
 * Helper class to provide an App instance and its associated Server instance.
 */
public class ServerAndApp {

	private Server server;
	private App app;
	
	/**
	 * Constructor.
	 * @param server The server.
	 * @param app The application.
	 */
	public ServerAndApp(Server server, App app) {
		this.server = server;
		this.app = app;
	}
	
	/**
	 * Terminates the app and the server.
	 */
	public void terminate() {
		app.terminate();
		server.terminate();
	}
	
	/**
	 * Gets the server.
	 * @return The server.
	 */
	public Server getServer() {
		return server;
	}
	
	/**
	 * Gets the app.
	 * @return The app.
	 */
	public App getApp() {
		return app;
	}
	
	
}